galasa-parent/dev.galasa.framework.api.authentication/src/main/java/dev/galasa/framework/api/authentication/internal/OidcProvider.java

```diff
@@ -175,7 +267,38 @@ private RSAPublicKey getRSAPublicKeyFromIssuer(String keyId, String issuer)
         return generatedPublicKey;
     }
 
-    public String getIssuer() {
-        return this.issuerUrl;
+    /**
+     * Returns the value of the "Location" HTTP header in a given HTTP response if
+     * one exists, returns null otherwise.
+     */
+    private String getLocationHeaderFromResponse(HttpResponse<String> response) {
+        HttpHeaders headers = response.headers();
+
+        Optional<String> locationHeader = headers.firstValue("location");
+        if (locationHeader.isPresent()) {
+            return locationHeader.get();
```

Comment:
I prefer a variable with the returned value being set and a single return statement.
Is it worth logging the returned value ?